_docs_integrate/error-codes.md

```diff
@@ -24,74 +24,163 @@ required_by:
+| {% include anchor a="error.transport.challenges.challengeTypeRequiresRelationship" %}                             | The challenge type 'Relationship' requires a Relationship for the challenge creation.                                                                                                                                                                           |
+| {% include anchor a="error.transport.datawallet.currentBiggerThanTarget" %}                                       | The current datawallet version is bigger than the target version.                                                                                                                                                                                               |
+| {% include anchor a="error.transport.datawallet.encryptedPayloadIsNoCipher" %}                                    | The given encrypted payload is no cipher.                                                                                                                                                                                                                       |
+| {% include anchor a="error.transport.datawallet.insufficientSupportedDatawalletVersion" %}                        | The current SupportedDatawalletVersion is too low to perform the requested operation. This means that the datawallet of the identity was upgraded by another device with a higher version. It is necessary to update the current device.                        |
+| {% include anchor a="error.transport.datawallet.unsupportedModification" %}                                       | A collection was recieved in a CacheChanged datawallet modification but is not supported in the current version of the library.                                                                                                                                 |
+| {% include anchor a="error.transport.devices.alreadyOnboarded" %}                                                 | The device that you try to create an onboarding code for has already been onboarded.                                                                                                                                                                            |
+| {% include anchor a="error.transport.devices.couldNotDeleteDevice" %}                                             | The device that you try to delete could not be deleted. Either the device is already onboarded or the backbone did not authorize deletion. [Get device onboarding info]({% link _docs_use-cases/use-case-transport-get-device-onboarding-info.md %})            |
```

Comment:
Isn't it necessary for a device to be onboarded before I can delete it?

Reply:
@jkoenig134 As you have implemented this I would like to talk with you regarding this, can you call me when you are back and have time?

Reply:
> Isn't it necessary for a device to be onboarded before I can delete it?

no, it's not possible to delete a device that was onboarded.

--- 

But - I'd not document any of the devices error codes as they will never occur in the connector.